user_test.go

```diff
@@ -19,6 +19,45 @@ func TestGetUserInfo(t *testing.T) {
 	require.Nil(t, err)
 	assert.Equal(t, PassageUserID, user.ID)
 }
+
+func TestGetUserInfoByIdentifier(t *testing.T) {
+	psg, err := passage.New(PassageAppID, &passage.Config{
+		APIKey: PassageApiKey,
+	})
+	require.Nil(t, err)
+
+	createUserBody := passage.CreateUserBody{
+		Email: RandomEmail,
```

Comment:
This test is only passing now because `RandomEmail` only generates lowercase letters. https://github.com/passageidentity/passage-go/blob/e4363a976dbcef6305a80fc5079e8cee391aae0f/passage_test.go#L29-L30

More test cases with uppercase characters in the identifier and searching for the lowercase counterpart would be good to cover the case-insensitivity.